builder/vsphere/supervisor/step_create_source.go

```diff
@@ -0,0 +1,306 @@
+//go:generate packer-sdc struct-markdown
+//go:generate packer-sdc mapstructure-to-hcl2 -type CreateSourceConfig
+
+package supervisor
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/hashicorp/packer-plugin-sdk/communicator"
+	"github.com/hashicorp/packer-plugin-sdk/multistep"
+	vmopv1alpha1 "github.com/vmware-tanzu/vm-operator-api/api/v1alpha1"
+)
+
+const (
+	DefaultSourceName  = "packer-vsphere-supervisor-built-source"
+	VMSelectorLabelKey = DefaultSourceName + "-selector"
+
+	StateKeySourceName              = "source_name"
+	StateKeyVMCreated               = "vm_created"
+	StateKeyVMServiceCreated        = "vm_service_created"
+	StateKeyVMMetadataSecretCreated = "vm_metadata_secret_created"
+)
+
+type CreateSourceConfig struct {
+	// Name of the the source virtual machine (VM) image.
+	ImageName string `mapstructure:"image_name" required:"true"`
+	// Name of the VM class that describes virtual hardware settings.
+	ClassName string `mapstructure:"class_name" required:"true"`
+	// Name of the storage class that configures storage-related attributes.
+	StorageClass string `mapstructure:"storage_class" required:"true"`
+
+	// Name of the source VM. Defaults to `packer-vsphere-supervisor-built-source`.
```

Comment:
Yes, it would cause a conflict as the source VM is a namespace-scoped resource in K8s cluster. I've changed it to append a random suffix for the default source name (similar to what `generateName` does in K8s).